APCAppCore/APCAppCore/Library/Categories/NewDataCollector/Data Sink/APCPassiveDataSink.m

```diff
@@ -0,0 +1,472 @@
+    id          dateString    = [date toStringInISO8601Format];
+    
+    if (dateString == nil)
+    {
+        dateString = [NSNull null];
+    }
+
+    infoDictionary = @{
+                       kIdentifierKey   : sinkIdentifier,
+                       kStartDateKey    : dateString
+                       };
+    
+    NSString*       infoJSON        = [infoDictionary JSONString];
+    
+    [APCPassiveDataSink createOrReplaceString:infoJSON toFile:infoFilePath];
+    
+    //Create data csv file
+    NSString*       rowString       = [[[self columnNames] componentsJoinedByString:@","] stringByAppendingString:@"\n"];
+    
+    [APCPassiveDataSink createOrAppendString:rowString toFile:csvFilePath];
+}
+
+- (NSDate*)datefromDateString:(NSString*)string
+{
+    NSDateFormatter* dateFormat = [[NSDateFormatter alloc] init];
+    
+    [dateFormat setDateFormat:@"yyyy-MM-dd'T'HH:mm:ssZZZZZ"];
+    
+    return [dateFormat dateFromString:string];
+}
+
++ (void)createOrAppendString:(NSString*)string toFile:(NSString*)path
+{
+    if (![[NSFileManager defaultManager] fileExistsAtPath:path])
+    {
+        [[string dataUsingEncoding:NSUTF8StringEncoding] writeToFile:path atomically:YES];
+    }
+    else
+    {
+        NSFileHandle* fileHandler = [NSFileHandle fileHandleForUpdatingAtPath:path];
+        
+        [fileHandler seekToEndOfFile];
+        [fileHandler writeData:[string dataUsingEncoding:NSUTF8StringEncoding]];
+        [fileHandler closeFile];
+    }
+}
+
+
++ (void)createOrReplaceString:(NSString*)string toFile:(NSString*)path
+{
+    if (![[NSFileManager defaultManager] fileExistsAtPath:path])
+    {
+        NSError* error = nil;
+        
+        if (![string writeToFile:path atomically:YES encoding:NSUTF8StringEncoding error:&error])
+        {
+            if (error)
+            {
+                APCLogError2(error);
+            }
+        }
+    }
+    else
+    {
+        NSError* error = nil;
+        if (![[NSFileManager defaultManager] removeItemAtPath:path error:&error])
+        {
+            if (error)
+            {
+                APCLogError2(error);
+            }
+        }
+        else
+        {
+            NSError* writeError = nil;
+            
+            if (![string writeToFile:path atomically:YES encoding:NSUTF8StringEncoding error:&error])
+            {
+                if (writeError)
+                {
+                    APCLogError2(writeError);
+                }
+            }
+        }
+    }
+}
+
++ (void)createFolderIfDoesntExist:(NSString*)path
+{
+    if (![[NSFileManager defaultManager] fileExistsAtPath:path])
+    {
+        NSError* folderCreationError = nil;
+        
+        if (![[NSFileManager defaultManager] createDirectoryAtPath:path
+                                       withIntermediateDirectories:YES
+                                                        attributes:@{ NSFileProtectionKey : NSFileProtectionCompleteUntilFirstUserAuthentication }
```

Comment:
Unconvinced that this is sufficient file protection for health data. Shouldn't protection be Complete, or CompleteUnlessOpen?

Of course once you do that there is the chance, at any time, that you might not be able to read the data if operating in the background.